### PR TITLE
Support Scala.js and further reorg (part3)

### PR DESCRIPTION
**Pull request by aslesarenko**
In this PR:
- [x] new `sdk` module configured
- [x] more tests moved to `core-lib` module for both JVM and JS execution
- [x] exported JS classes moved to `sdk`
- [x] importing of sigmajs-crypto-facade is reconfigured
- [x] EIP3 addresses implementation (with related code) moved from ergo-wallet repo to `sdk` module
- [x] some tx signing code moved from Appkit to `sdk`

### CODE REVIEW

File: interpreter/js/src/main/scala/sigmastate/crypto/SigmaJsCryptoFacade.scala [hunk 1]
```diff
@@ -106,6 +106,18 @@ object CryptoFacadeJs extends js.Object {
...
+
+  /** Computes HMAC-SHA512 hash of the given data using the specified key.
+    *
+    * @param key  the secret key used for hashing
+    * @param data the input data to be hashed
+    * @return a HMAC-SHA512 hash of the input data
+    */
+  def hashHmacSHA512(key: Uint8Array, data: Uint8Array): Uint8Array = js.native
+
+  def generatePbkdf2Key(
```
Comment (kushti) [inline interpreter/js/src/main/scala/sigmastate/crypto/SigmaJsCryptoFacade.scala:118]: scaladoc missed

File: interpreter/js/src/main/scala/sigmastate/crypto/Platform.scala [hunk 2]
```diff
@@ -28,6 +29,14 @@ object Platform {
     jsShorts.toArray[Short].map(x => x.toByte)
   }
 
+  def bytesToJsShorts(bytes: Array[Byte]): js.Array[Short] = {
```
Comment (kushti) [inline interpreter/js/src/main/scala/sigmastate/crypto/Platform.scala:32]: ScalaDoc missed here and for all other new methods in the class

**Review by kushti — APPROVED** [review]

File: interpreter/js/src/main/scala/sigmastate/crypto/SigmaJsCryptoFacade.scala (hunk 1 above)
Comment (aslesarenko) [reply interpreter/js/src/main/scala/sigmastate/crypto/SigmaJsCryptoFacade.scala:118]: fixed

File: interpreter/js/src/main/scala/sigmastate/crypto/Platform.scala (hunk 2 above)
Comment (aslesarenko) [reply interpreter/js/src/main/scala/sigmastate/crypto/Platform.scala:53]: fixed

**Pull request merged**